### ISSUE TITLE
Support for uploading any file type

### ISSUE THREAD

**Issue by elberre57** [hoarder-app/hoarder#601]
### Describe the feature you'd like

A feature that would allow users to upload any file type. If a preview is not (yet) supported, the file name and a download button would be displayed instead, allowing users to download the file.

### Describe the benefits this would bring to existing Hoarder users

The feature would make the app more flexible for different use cases. Users could store and organize the files they need for their workflows all in one place. It could also be used to quickly share and access files across multiple devices.

### Can the goal of this request already be achieved via other means?

The only workaround at the moment is to change the file extension to .jpg and upload it. Later, when you need to download it, you can download the 'image' and change its file extension back to the original.

### Have you searched for an existing open/closed issue?

- [X] I have searched for existing issues and none cover my fundamental request

### Additional context

I created this mockup to give you an idea of how such a feature could look.
![mockup](https://github.com/user-attachments/assets/a3d737be-3552-453c-ae7e-fc0b440d8e91)

**Comment by KingHyol** [comment]
I also support this content.
If documents can be kept like the function of paperless-ngx, it can be used as a more versatile application

**Comment by CrazyWolf13** [comment]
yes I really hope this makes it into karakeep, I use paperless as well, but that serves a different use case.

For karakeep something like this came to my mind:

I save a blog that mentions a hiking tour, as user I want to upload the gpx file I created for the tour, but karakeep sadly refuses it.
